StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -691,25 +654,20 @@ extension PaymentSheet {
                     }
                 } else {
                     // Client-side confirmation
-                    var confirmableIntent: Intent.Confirmable
-
-                    switch deferredIntentContext.intentConfig.mode {
-                    case .payment:
-                        confirmableIntent = .payment(clientSecret: clientSecret, paymentIntent: nil)
-                    case .setup:
-                        confirmableIntent = .setup(clientSecret: clientSecret, setupIntent: nil)
-                    }
-
+                    let intent = try await deferredIntentContext.configuration.apiClient.retrieveIntent(for: deferredIntentContext.intentConfig,
+                                                                                                        withClientSecret: clientSecret)
                     confirm(configuration: deferredIntentContext.configuration,
                             authenticationContext: deferredIntentContext.authenticationContext,
-                            confirmableIntent: confirmableIntent,
+                            intent: intent,
                             paymentOption: deferredIntentContext.paymentOption,
                             paymentHandler: deferredIntentContext.paymentHandler,
                             completion: deferredIntentContext.completion)
                 }
 
             } catch {
-                deferredIntentContext.completion(.failed(error: error))
+                DispatchQueue.main.async {
```

Comment:
This is all we need to resolve the threading crash

Reply:
I've also manually verified that when the merchant invokes the `intentCreationCallback` from the background thread that all tests pass.

Reply:
Why only call the completion block on the main thread in the error case, and not the success cases?

After the completion block is called here, the control flow will eventually return to view controllers or PaymentSheet.FlowController etc. To be a good citizen to the rest of the codebase, it should always return control on the main thread. 

As a shortcut, you could take advantage of `DeferredIntentContext` to wrap the completion block in a `DispatchQueue.main.async` call in its initializer.